src/atomvm_escriptize_provider.erl

```diff
@@ -0,0 +1,456 @@
+%%
+%% Copyright (c) 2025 Paul Guyot <pguyot@kallisys.net>
+%% All rights reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%
+% SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+%
+
+-module(atomvm_escriptize_provider).
+
+-behaviour(provider).
+
+-export([init/1, do/1, format_error/1]).
+
+-include_lib("kernel/include/file.hrl").
+
+-define(PROVIDER, escriptize).
+-define(DEPS, [packbeam]).
+-define(OPTS, [
+    {atomvm_binary, $b, "atomvm_binary", string, "Path to AtomVM binary (default: which AtomVM)"},
```

Comment:
if we default to `which atomvm`, and use the launcher script from an installed AtomVM then we don't need to have atomvmlib supplied either. For those not using an installed AtomVM they will need to supply both.